rust/src/lib/nm/connection.rs

```diff
@@ -251,39 +213,55 @@ pub(crate) fn get_port_nm_conns<'a>(
     ret
 }
 
-pub(crate) fn gen_nm_connection(
-    iface_name: &str,
-    uuid: &str,
-    nm_iface_type: &str,
-    ctrl_name: Option<&str>,
-    nm_ctrl_type: Option<&str>,
-    is_controller: bool,
-) -> NmConnection {
-    let mut nm_conn = NmConnection::new();
-
-    // OVS port already has it own prefix
-    let conn_name = if nm_iface_type == "ovs-bridge" {
-        format!("ovs-br-{}", iface_name)
-    } else if nm_iface_type == "ovs-interface" {
-        format!("ovs-iface-{}", iface_name)
+pub(crate) fn gen_nm_conn_setting(
+    iface: &Interface,
+    nm_conn: &mut NmConnection,
+) -> Result<(), NmstateError> {
+    let mut nm_conn_set = if let Some(cur_nm_conn_set) = &nm_conn.connection {
```

Comment:
Can be:
```
let mut nm_conn_set = nm_conn.connection.as_ref().cloned().unwrap_or_default();
```

Reply:
Missed comment?

Reply:
This one cannot use `unwrap_or_default` as we need to setup many stuff for new connection instead of default.